### PR TITLE
Security level datums

### PR DESCRIPTION
**Pull request by adamsong**
# Document the changes in your pull request

Steals tgstation/tgstation#67949 from tgstation

# Why is this good for the game?
Code better

# Testing
It compiles
image

### CODE REVIEW

**Comment by ToasterBiome** [comment]
reopen when work continues

File: code/modules/antagonists/blob/overmind.dm [hunk 1]
```diff
@@ -115,7 +115,7 @@ GLOBAL_LIST_EMPTY(blob_nodes)
 	else if(!victory_in_progress && (blobs_legit.len >= blobwincount))
 		victory_in_progress = TRUE
 		priority_announce("Biohazard has reached critical mass. Station loss is imminent.", "Biohazard Alert")
-		set_security_level("delta")
+		SSsecurity_level.set_level("delta")
```
Comment (JohnFulpWillard) [inline code/modules/antagonists/blob/overmind.dm]: Can you swap this for the define

File: code/modules/antagonists/traitor/equipment/Malf_Modules.dm [hunk 2]
```diff
@@ -232,7 +232,7 @@ GLOBAL_LIST_INIT(malf_modules, subtypesof(/datum/AI_Module))
 	if(owner.stat == DEAD)
 		return
 	priority_announce("Hostile runtimes detected in all station systems, please deactivate your AI to prevent possible damage to its morality core.", "Anomaly Alert", ANNOUNCER_AIMALF)
-	set_security_level("delta")
+	SSsecurity_level.set_level("delta")
```
Comment (JohnFulpWillard) [inline code/modules/antagonists/traitor/equipment/Malf_Modules.dm]: Here too

File: code/modules/mob/living/silicon/ai/death.dm [hunk 3]
```diff
@@ -47,7 +47,7 @@
 
 /mob/living/silicon/ai/proc/ShutOffDoomsdayDevice()
 	if(nuking)
-		set_security_level("red")
```
Comment (JohnFulpWillard) [inline code/modules/mob/living/silicon/ai/death.dm:50]: And here

File: code/modules/power/singularity/narsie.dm [hunk 4]
```diff
@@ -94,15 +94,15 @@
 		return
 	priority_announce("Simulations on acausal dimensional event complete. Deploying solution package now. Deployment ETA: 30 SECONDS.","Central Command Higher Dimensional Affairs")
 	sleep(5 SECONDS)
-	set_security_level("delta")
+	SSsecurity_level.set_level("delta")
```
Comment (JohnFulpWillard) [inline code/modules/power/singularity/narsie.dm]: And here

**Review by JohnFulpWillard — COMMENTED** [review]
I'm going to sleep so this isn't reviewing code but just things I saw scrolling while on mobile

File: code/modules/security_levels/security_level_datums.dm [hunk 5]
```diff
@@ -77,45 +94,64 @@
...
 	shuttle_call_time_mod = ALERT_COEFF_EPSILON
+	disable_night_mode = TRUE
+	pod_access = TRUE
+	emergency_doors = TRUE
+	allow_cryo = FALSE
+
+/datum/security_level/epsilon/on_activate(previous_level)
+	send_to_playing_players(span_notice("You get a bad feeling as you hear the Epsilon alert siren."))
+
+/**da
```
Comment (JohnFulpWillard) [inline code/modules/security_levels/security_level_datums.dm]:
```suggestion
/**
```

File: code/modules/security_levels/security_levels.dm [hunk 6]
```diff
@@ -1,167 +1,155 @@
...
-			return SEC_LEVEL_BLUE
-		if("red")
-			return SEC_LEVEL_RED
-		if("gamma")
-			return SEC_LEVEL_GAMMA
-		if("epsilon")
-			return SEC_LEVEL_EPSILON
-		if("delta")
-			return SEC_LEVEL_DELTA
+// TODO: Delete this, just leaving it while I make sure everything works
```
Comment (JohnFulpWillard) [inline code/modules/security_levels/security_levels.dm]: Delete this plea

File: code/modules/security_levels/security_level_datums.dm [hunk 7]
```diff
@@ -77,45 +94,64 @@
...
+	disable_night_mode = TRUE
+	pod_access = TRUE
+	emergency_doors = TRUE
+	allow_cryo = FALSE
 
 /**
  * EPSILON
  *
- * Station destruction is here, and you probably caused it, you monster.
+ * [REDACTED]
```
Comment (JohnFulpWillard) [inline code/modules/security_levels/security_level_datums.dm]: What is this change for lol

Comment (adamsong) [reply code/modules/security_levels/security_level_datums.dm]: The original comment is for delta, not epsilon

Comment (JohnFulpWillard) [reply code/modules/security_levels/security_level_datums.dm]: can you describe what it is for then

**Review by JohnFulpWillard — APPROVED** [review]

**Review by ToasterBiome — APPROVED** [review]

**Pull request merged**